src/client.ts

```diff
@@ -231,20 +236,23 @@ in an environmental variable named FAUNA_SECRET or pass it to the Client\
         Authorization: `Bearer ${this.#clientConfiguration.secret}`,
         // WIP - typecheck should be user configurable, but hard code for now
         "x-typecheck": "false",
-        // WIP - presently core will default to tagged; hardcode to simple for now
-        // until we get back to work on the JS driver.
-        "x-format": "simple",
-        // include per-request headers
+        "x-format": this.#clientConfiguration.format,
       };
       this.#setHeaders(
         { ...this.clientConfiguration, ...queryRequest },
         headers
       );
 
+      const isTaggedFormat =
+        this.#clientConfiguration.format === "tagged" ||
+        queryRequest.format === "tagged";
```

Comment:
I updated this. The logic is the same, but I still referenced the client and request configurations. It could be slightly more terse if we used the defined header value, but then we'd have to get `headers["x-format"]`, which is not typed (and also a bit uglier IMO).